webapp/src/main.js

```diff
@@ -43,9 +44,14 @@ async function init () {
 
 	const token = new URLSearchParams(router.currentRoute.hash.substr(1)).get('token')
 	if (token) {
-		localStorage.token = token
+		if (!config.externalAuthUrl) {
+			localStorage.token = token
+		}
 		router.replace(router.currentRoute.path)
 		store.dispatch('login', {token})
+	} else if (config.externalAuthUrl) {
+		location = config.externalAuthUrl
```

Comment:
My personal taste would be to prefer `window.location` to make clear this is not a local variable of some kind